src/test/scala/com/ubirch/util/elasticsearch/TestUtils.scala

```diff
@@ -27,46 +29,37 @@ trait TestUtils
   protected var esMappingImpl: TestEsMappingImpl = _
   protected var simpleClient: TestEsSimpleClient = _
   protected var bulkClient: TestEsBulkClient = _
-  protected var client: RestHighLevelClient = _
+  protected var client: ElasticsearchAsyncClient = _
 
-  class TestEsSimpleClient(client: RestHighLevelClient) extends EsSimpleClientBase {
-    override val esClient: RestHighLevelClient = client
+  class TestEsSimpleClient(client: ElasticsearchAsyncClient) extends EsSimpleClientBase {
+    override val esClient: ElasticsearchAsyncClient = client
   }
 
-  class TestEsBulkClient(client: RestHighLevelClient) extends EsBulkClientBase {
-    override val esClient: RestHighLevelClient = client
+  class TestEsBulkClient(client: ElasticsearchAsyncClient) extends EsBulkClientBase {
+    override val esClient: ElasticsearchAsyncClient = client
   }
 
-  class TestEsHighLevelClient(testHost: String, httpPort: Int) extends EsHighLevelClient {
+  class TestEsAsyncClient(testHost: String, httpPort: Int) extends EsAsyncClient {
     override lazy val port: Int = httpPort
     override lazy val host: String = testHost
   }
 
-  class TestEsMappingImpl(client: RestHighLevelClient) extends EsMappingTrait {
-    override val esClient: RestHighLevelClient = client
+  class TestEsMappingImpl(client: ElasticsearchAsyncClient) extends EsMappingTrait {
+    override val esClient: ElasticsearchAsyncClient = client
 
-    override val indexesAndMappings: Map[String, String] =
-      Map(docIndex ->
-        s"""{
-           |    "properties" : {
-           |      "id" : {
-           |        "type" : "keyword"
-           |      },
-           |      "hello" : {
-           |        "type" : "keyword"
-           |      },
-           |      "value" : {
-           |        "type" : "integer"
-           |      }
-           |    }
-           |}""".stripMargin)
+    override val indexesAndMappings =
+      Map(docIndex -> Map(
+        "id" -> new KeywordProperty.Builder().build()._toProperty(),
+        "hello" -> new KeywordProperty.Builder().build()._toProperty(),
+        "value" -> new KeywordProperty.Builder().build()._toProperty()
+      ))
   }
 
   override val container: ElasticsearchContainer =
```

Comment:
As long as I understand, each Spec extending the trait starts running a container.
So running container 4 times in total?

Reply:
So much better now. :)